### PR TITLE
Upload the model with push_to_hub in examples

### PR DESCRIPTION
**Pull request by leshanbog**
Pushing model and optimizer checkpoints to HuggingFace Hub using their API

### CODE REVIEW

**Review by justheuristic — APPROVED** [review]
everything appears to be in order

@yhn112 
Let's test this in one of the internal runs (we could take an existing run and add an auxiliary uploader @ QYP, then see if it can upload 2 consecutive checkpoints)

File: examples/albert/run_first_peer.py [hunk 1]
```diff
@@ -52,6 +52,10 @@ class CoordinatorArguments(BaseTrainingArguments):
         default=None,
         metadata={"help": "Path to HuggingFace repo in which coordinator will upload the model and optimizer states"}
     )
+    repo_url: Optional[str] = field(
+        default=None,
+        metadata={"help": "Url to HuggingFace repo in which coordinator will upload the model and optimizer states"}
```
Comment (mryab) [inline examples/albert/run_first_peer.py:57]:
```suggestion
        metadata={"help": "URL to Hugging Face repository to which the coordinator will upload the model and optimizer states"}
```

File: examples/albert/run_first_peer.py [hunk 2]
```diff
@@ -118,17 +124,13 @@ def is_time_to_upload(self):
...
         self.previous_timestamp = time.time()
-        try:
-            subprocess.run("git add --all", shell=True, check=True, cwd=self.repo_path)
-            current_step = self.collaborative_optimizer.collaboration_state.optimizer_step
-            subprocess.run(f"git commit -m 'Step {current_step}, loss {current_loss:.3f}'",
-                           shell=True, check=True, cwd=self.repo_path)
-            subprocess.run("git push", shell=True, check=True, cwd=self.repo_path)
-        except subprocess.CalledProcessError as e:
-            logger.warning("Error while uploading model:", e.output)
+        logger.info('Start uploading model to hub')
```
Comment (mryab) [inline examples/albert/run_first_peer.py:130]:
```suggestion
        logger.info('Start uploading model to Hub')
```

File: examples/albert/run_first_peer.py [hunk 3]
```diff
@@ -118,17 +124,13 @@ def is_time_to_upload(self):
...
+        self.model.push_to_hub(repo_name=self.repo_path, repo_url=self.repo_url,
+                               commit_message=f'Step {current_step}, loss {current_loss:.3f}')
+        logger.info('Finish uploading model to hub')
```
Comment (mryab) [inline examples/albert/run_first_peer.py:133]:
```suggestion
        logger.info('Finish uploading model to Hub')
```

File: examples/albert/run_first_peer.py [hunk 4]
```diff
@@ -118,17 +124,13 @@ def is_time_to_upload(self):
...
         self.previous_timestamp = time.time()
-        try:
-            subprocess.run("git add --all", shell=True, check=True, cwd=self.repo_path)
-            current_step = self.collaborative_optimizer.collaboration_state.optimizer_step
-            subprocess.run(f"git commit -m 'Step {current_step}, loss {current_loss:.3f}'",
-                           shell=True, check=True, cwd=self.repo_path)
-            subprocess.run("git push", shell=True, check=True, cwd=self.repo_path)
-        except subprocess.CalledProcessError as e:
-            logger.warning("Error while uploading model:", e.output)
+        logger.info('Start uploading model to Hub')
```
Comment (mryab) [inline examples/albert/run_first_peer.py:130]:
```suggestion
        logger.info('Started uploading model to Hub')
```

File: examples/albert/run_first_peer.py [hunk 5]
```diff
@@ -118,17 +124,13 @@ def is_time_to_upload(self):
...
+        self.model.push_to_hub(repo_name=self.repo_path, repo_url=self.repo_url,
+                               commit_message=f'Step {current_step}, loss {current_loss:.3f}')
+        logger.info('Finish uploading model to Hub')
```
Comment (mryab) [inline examples/albert/run_first_peer.py:133]:
```suggestion
        logger.info('Finished uploading model to Hub')
```

**Review by mryab — APPROVED** [review]

**Pull request merged**